core/src/sprocket-rating/sprocket-rating.service.ts

```diff
@@ -0,0 +1,61 @@
+import {Injectable, Logger} from "@nestjs/common";
+
+import type {MLE_PlayerStatsCore} from "../database/mledb";
+
+export interface SprocketRating {
+    opi: number;
+    dpi: number;
+    gpi: number;
+}
+
+@Injectable()
+export class SprocketRatingService {
+    private logger = new Logger(SprocketRatingService.name);
+
+    calcSprocketRating(core: Partial<MLE_PlayerStatsCore>): SprocketRating {
```

Comment:
Not sure how much this would change things; but can we break the dependence on the MLEDB type for this?

Reply:
Really it would either drastically change the API for this method (by passing in each field separately) or mandate yet another payload type specific to this calculation. We can probably think that through a bit better down the line.

Reply:
Type added.